src/rapids_singlecell/decoupler_gpu/_method_mlm.py

```diff
@@ -102,6 +128,8 @@ def run_mlm(
             Whether to show progress.
         use_raw
             Use raw attribute of mat if present.
+        pre_load
+            Whether to pre-load the data into memory. This can be faster for small datasets.
```

Comment:
You don’t handle `None` yet, neither in code nor in the docs. Should be something like this in the public API:

```py
if pre_load is None:
    pre_load = isinstance(m, CupyArray)
```

and then the private APIs (`mlm`, `wsum`) just accept `pre_load: bool` .

Reply:
Its still in early development I have to rewrite some other things and I'll handle it than